ui: cluster overview layout tweaks

couchand opened the pull request:
Some tweaks to the cluster overview layout for narrow and wide screens, as well as better handling when the metrics have many-digit numbers.

<img width="767" alt="screen shot 2018-04-17 at 1 35 41 pm" src="https://user-images.githubusercontent.com/793969/38886599-6f3704ee-4244-11e8-9ff9-bdf476176566.png">
<img width="1021" alt="screen shot 2018-04-17 at 1 35 57 pm" src="https://user-images.githubusercontent.com/793969/38886600-6f5bec50-4244-11e8-9ace-9cf88cc3ddd1.png">
<img width="1438" alt="screen shot 2018-04-17 at 1 36 18 pm" src="https://user-images.githubusercontent.com/793969/38886601-6f7d2af0-4244-11e8-8e9d-d228cc72a4c4.png">
<img width="1920" alt="screen shot 2018-04-17 at 1 36 33 pm" src="https://user-images.githubusercontent.com/793969/38886602-6fa30b80-4244-11e8-9e7e-7f49b343e1f7.png">

Avoiding collision:
<img width="195" alt="screen shot 2018-04-16 at 5 48 37 pm" src="https://user-images.githubusercontent.com/793969/38837026-6fcebeb6-419e-11e8-914b-4cb12efc365b.png">

Fixes: #23669
Fixes: #24158
Fixes: #24844

couchand committed:
ui: ensure overview metrics don't collide

Fixes: #23669
Release note: None

couchand committed:
ui: responsive layouts for cluster overview

This adds several secondary layouts for the cluster overview
intended for narrow and wide screens.  The narrow layout removes
the capacity chart and puts each metric on a row by itself.
The wide layouts grow font sizes and spacing appropriately.

Fixes: #24158
Release note (admin ui change): The cluster overview now adjusts
to take the advantage of the size of the screen.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/24849)
<!-- Reviewable:end -->

couchand commented:
Needs design review.  cc @josueeee

vilterp commented on pkg/ui/src/views/cluster/containers/clusterOverview/cluster.styl:
```diff
@@ -31,6 +31,17 @@
+    @media screen and (max-width: 960px)
+      grid-template-columns 6fr 8fr 6fr 8fr
+      grid-template-rows repeat(3, auto) 16px repeat(3, auto) 16px repeat(3, auto)
+      grid-template-areas "cap-t cap-t cap-t cap-t" "cap-m cap-c cap-c cap-c" "cap-a1 cap-a2 cap-a3 cap-a4" ". . . ." "live-t live-t live-t ." "live-a live-b live-c ." "live-1 live-2 live-3 ." ". . . ." "rep-t rep-t rep-t ." "rep-a rep-b rep-c ." "rep-1 rep-2 rep-3 ."
```
whoah, didn't see these when you first implemented this. nothing you can do since it's just the CSS spec, but not the easiest to read!

couchand replied:
ugh stylus ruins this.  it looks much nicer in plain css (ok, still weird, but at least you can multi-line it)

vilterp reviewed (approved):
LGTM. Schnazzy!

josueeee commented:
Please review the Figma file for breakpoint layout suggestions: https://www.figma.com/file/R9LIMJmhrzRXDBtfUGDTs3mh/02_Overview?node-id=16%3A2

couchand committed:
ui: allow internal scrolling on node list

Previously, when the screen was too narrow for the node list,
the overflow would just be hidden.  This allows internal
scrolling when that happens.

Fixes: #24844
Release note (admin ui change): Fixes a bug where the node list
could get cut off.

couchand commented:
@josueeee I think I addressed most of your suggestions, let me know what you think.

couchand commented:
@vilterp take another look and feel free to complain about all random media queries

vilterp reviewed (approved):
Still LGTM. Only nit is that
there's margin on the left: ![image](https://user-images.githubusercontent.com/7341/38887805-d2e06690-4247-11e8-89d7-c44a8951e57d.png) but not the right: ![image](https://user-images.githubusercontent.com/7341/38887795-cd9bf032-4247-11e8-95ab-1a9224579e6c.png)

Should have margin on both or neither I think.

couchand commented:
I can't seem to figure out what's going on with that right padding, the internet seems to think a wrapper div should fix it but I can't seem to make that work.  Punting to #24882.

Further tweaks can reasonably be made in follow-up work.

bors r+

couchand commented:
Ha!  Way to go Craig!

bors r+

vilterp commented:
Looks like `TestGossipRestart` is failing :(

couchand commented:
i hope the layout isn't killing that test :joy:/:sob:
